### PR TITLE
Move cose & cbor support to common

### PR DESCRIPTION
**Pull request by robinbryce**
In support of merklelog proof & receipt format. Given the prevalence of cbor & cose in scitt and IETF it seems right to have this in common

Additional:

* Refactor the Verification methods to avoid a circular dependency with the did package

[AB#8336](https://dev.azure.com/jitsuin/0629f48c-3979-4bbc-9026-cb06b3dfd0ae/_workitems/edit/8336)

### CODE REVIEW

File: cose/cose.go [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+	kidBytes, ok := kid.([]byte)
+	if !ok {
+		logger.Sugar.Infof("kidFromProtectedHeader: kid from protected header is not []byte: %v", err)
+		return "", &ErrUnexpectedProtectedHeaderType{label: cose.HeaderLabelKeyID, expectedType: "[]byte", actualType: reflect.TypeOf(kid).String()}
+	}
+
+	return string(kidBytes), nil
+}
+
+type publicKeyProvider interface {
```
Comment (honourfish) [inline cose/cose.go:290]: nice!

**Review by honourfish — APPROVED** [review]

File: cbor/codec.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package cbor
```
Comment (eccles) [inline cbor/codec.go:1]: The name of this package and the name of the cbor/v2 package are the same which is confusing perhaps

File: cbor/codec.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+import "github.com/fxamacker/cbor/v2"
+
+// CBORCodec encode decode
+type CBORCodec struct {
```
Comment (eccles) [inline cbor/codec.go:6]: stuttering by repeating CBOR is discouraged - why not just Codec. USer code will refer to this as cbor.Codec which is clear enough?

File: cose/cose.go [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"errors"
+	"io"
+	"reflect"
+
+	dtcbor "github.com/datatrails/go-datatrails-common/cbor"
+	"github.com/datatrails/go-datatrails-common/logger"
+	"github.com/fxamacker/cbor/v2"
```
Comment (eccles) [inline cose/cose.go:12]: why do we need to import the external dependency cbor  as well as the go-datatrails-common cbor?

File: cose/cwt_claims.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+//		/COSE_Key/ 1 :{
+//			/kty/ 1 : /EC2/ 2,
+//			/crv/ -1 : /P-256/ 1,
+//			/x/ -2 : h'd7cc072de2205bdc1537a543d53c60a6acb62eccd890c7fa27c9
+//					   e354089bbe13',
+//			/y/ -3 : h'f95e1d4b851a2cc80fff87d8e23f22afb725d535e515d020731e
+//					   79a3b4e47120'
+//		   }
+//		 }
+func CNFCoseKey(cwtClaimsMap map[interface{}]interface{}) (CoseKey, error) {
```
Comment (eccles) [inline cose/cwt_claims.go:47]: any instead of interface{}

File: cose/errors.go [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+// ErrKeyValueError occurs when the key has unexpected values
+type ErrKeyValueError struct {
+
+	// field is the field that is unexpected
+	field string
+
+	// value is the value of the field
+	value interface{}
```
Comment (eccles) [inline cose/errors.go:63]: any

File: cose/cwt_claims.go (hunk 5 above)
Comment (robinbryce) [reply cose/cwt_claims.go:47]: I just moved this code

File: cose/errors.go (hunk 6 above)
Comment (robinbryce) [reply cose/errors.go:63]: I just moved this code

File: cose/cose.go (hunk 4 above)
Comment (robinbryce) [reply cose/cose.go:12]: they do different things. There is no need to hid the actual cbor under a datatrails cbor. we are not abstracting. and I don't think the attempts we have made in other places to do that have worked out well.

**Pull request merged**